zest/releaser/bzr.py

```diff
@@ -40,24 +40,25 @@ def tag_url(self, version):
         return version
 
     def cmd_diff(self):
-        return 'bzr diff'
+        return [u'bzr', u'diff']
 
     def cmd_commit(self, message):
-        return 'bzr commit -v -m "%s"' % message
+        return [u'bzr', u'commit', u'-v', u'-m', message]
 
     def cmd_diff_last_commit_against_tag(self, version):
-        return "bzr diff -r tag:%s..-1" % version
+        return [u"bzr", u"diff", u"-r", u"tag:{0}..-1".format(version)]
 
     def cmd_log_since_tag(self, version):
-        return "bzr log -r tag:%s..-1" % version
+        return [u"bzr", u"log", u"-r", u"tag:{0}..-1".format(version)]
 
     def cmd_create_tag(self, version):
-        return 'bzr tag %s' % version
+        return [[u'bzr', u'tag', version]]
```

Comment:
Another VCS needs to return multiple commands from this method. It's better
to have a consistent signature than returning mixed results depending on
which VCS one is using.
On 2 Jun 2015 7:16 am, "Marius Gedminas" <notifications@github.com> wrote:

> In zest/releaser/bzr.py
> <https://github.com/zestsoftware/zest.releaser/pull/97#discussion_r31495087>
> :
>
> >
> >      def cmd_create_tag(self, version):
> > -        return 'bzr tag %s' % version
> > +        return [[u'bzr', u'tag', version]]
>
> Why the double nesting? Other commands don't do that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zestsoftware/zest.releaser/pull/97/files#r31495087>.
>